text_mining.py

```diff
@@ -0,0 +1,357 @@
+"""
+Text_Mining by Seungin Lyu
+Conducts sentiment anaylsis of all tweets of EPL players listed by
+"Preimer Leauge Players"(public BBC Tweeter list) who have unprotected
+timelines. The results for each indivual players are groupd by clubs to create
+club sentiment anaylsis data for each of the EPL clubs.
+"""
+
+#  Data Sources
+import twitter
+import wikipedia
+from bs4 import BeautifulSoup
+import warnings
+
+
+# Text Analyzer
+from vaderSentiment.vaderSentiment import SentimentIntensityAnalyzer
+
+import doctest
+import datetime
+import re
+from pickle import dump, load
+from os.path import exists
+from os import mkdir
+
+
+def get_credentials(filename):
+    """
+    Returns credentials for twitter api from file with filename
+    """
+    f = open('keys.txt')
+    keys = f.readlines()
+    # codes from Oliver Steele's Piazza Class notes : keeping secrets
+    credentials = []
+    try:
+        for line in keys:
+            credentials.append(line.strip())
+    finally:
+        f.close()
+    return credentials
+
+
+def save_epl_members_list(filename, api):
+    """
+    Retrieves sequence of twitter.user.User instances of
+    "Premier League Players" Twitter list(public list by BBC) as a list
+    (This list might not be completely up to date, so not every member
+    of this list still players in the EPL)
+    Then Pickles the twitter user instance for each player.
+    """
+    epl_players = api.GetListMembers(None, 'premier-league-players',
+                                     None, 'BBCSport')
+    f = open(filename, 'wb')
+    dump(len(epl_players), f)  # saves the number of players
+    try:
+        for each_player in epl_players:
+            dump(each_player, f)
+    finally:
+        f.close()
+
```

Comment:
It's probably a good idea to separate data fetching functionality and data analysis functionality into two different files.